atc/worker/pool.go

```diff
@@ -151,16 +142,23 @@ func (pool *pool) findWorkerFromStrategy(
 		return nil, err
 	}
 
+	var combinedError error
 	for _, candidate := range orderedWorkers {
 		err := strategy.Pick(logger, candidate, containerSpec)
 
-		if err != nil {
-			logger.Debug("candidate-worker-rejected-during-selection", lager.Data{"reason": err.Error()})
-		} else {
+		if err == nil {
 			return candidate, nil
 		}
+
+		formattedError := fmt.Errorf("worker: %s, error: %v", candidate.Name(), err)
+		if combinedError == nil {
+			combinedError = formattedError
+		} else {
+			combinedError = fmt.Errorf("%v\n%v", combinedError, formattedError)
+		}
```

Comment:
Not asking you to change it, but just for future reference, you can replace this with 
```go
combinedError = multierror.Append(combinedError, formattedError)
```

using [hashicorp/go-multierror](https://github.com/hashicorp/go-multierror) (which we already happen to depend on)